google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+| Configuration | String | Name of the configuration to use. | | x
+| Ordering Key a| String |  If not empty, the ordering key identifies related messages in which the publish order must be respected. This key is respected when the subscriber enables message ordering. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration used to execute this component. |  | x
+| Project ID a| String |  Globally unique identifier for your project. |  |
+| Topic Name a| String |  Project-wide unique identifier for the topic. |  |
+| Message a| Binary |  Message content published to the topic. If not empty, the message must contain at least one attribute. |  |
+| Attributes a| Object |  Attributes for this message. If empty, the message must contain non-empty data. Can be used to filter messages on the subscription. |  |
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+| Type | String
+|===
+
+=== For Configurations
+
+* <<Config>>
+
+=== Throws
+
+* PUBSUB:BAD_GATEWAY
+* PUBSUB:CANCELLED
+* PUBSUB:CONNECTIVITY
+* PUBSUB:DEADLINE_EXCEEDED
+* PUBSUB:FAILED_PRECONDITION
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:MAX_OUTSTANDING_BATCH_SIZE_REACHED
+* PUBSUB:MAX_OUTSTANDING_ELEMENT_COUNT_REACHED
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:RETRY_EXHAUSTED
+* PUBSUB:UNAVAILABLE
+
+
+[[SeekMessages]]
+== Seek Messages
+`<pubsub:seek-messages>`
+
+
+Supports bulk acknowledging or un-acknowledging of messages to a given snapshot or point of time based on the seekTarget provided. Seeking to a time marks every message received by Pub/Sub before the time as acknowledged, and all messages received after the time as unacknowledged. You can replay and reprocess previously acknowledged messages when seeking to a time in the past or purge messages when seeking to a time in the future.
+
+
+If you seek to a timestamp using a subscription with a filter, the Pub/Sub service redelivers only the messages in the timestamp that match the filter of the subscription making the seek request. The snapshot feature enables you to capture the message acknowledgment state of a subscription. Once a snapshot is created, it retains:
+
+* all messages that were unacknowledged in the source subscription at the time of the snapshot's creation.
+* any messages published to the topic thereafter. You can replay these unacknowledged messages by using a snapshot to seek to any of the topic's subscriptions.
+
+
+If you seek to a snapshot using a subscription with a filter, the Pub/Sub service redelivers only the messages in the snapshot that match the filter of the subscription making the seek request.
+
```

Comment:
Once a snapshot is created, it retains all messages that were unacknowledged in the source subscription at the time of the snapshot's creation and any messages published to the topic thereafter. You can replay these unacknowledged messages by using a snapshot to seek to any of the topic's subscriptions.


If you seek to a snapshot using a subscription with a filter, the Pub/Sub service redelivers only the messages in the snapshot that match the filter of the subscription making the seek request.